### PR TITLE
s3rs: init at 0.4.8

### PR DESCRIPTION
**Pull request by yanganto**
###### Motivation for this change
Add s3 tools

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124884` at 01f1f399 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/7bee23a03218c63f5f08e885734a2adf">s3rs</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/tools/networking/s3rs/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, rustPlatform, python3, perl, openssl, Security, fetchFromGitHub, pkg-config }:
+rustPlatform.buildRustPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/s3rs/default.nix:2]:
```suggestion
{ lib, stdenv, rustPlatform, python3, perl, openssl, Security, fetchFromGitHub, pkg-config }:

rustPlatform.buildRustPackage rec {
```

File: pkgs/tools/networking/s3rs/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  pname = "s3rs";
+  version = "0.4.8";
+
+  src = fetchFromGitHub {
+    owner = "yanganto";
+    repo = "${pname}";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/s3rs/default.nix:9]:
```suggestion
    repo = pname;
```

File: pkgs/tools/networking/s3rs/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "s3rs";
+  version = "0.4.8";
+
+  src = fetchFromGitHub {
+    owner = "yanganto";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-qBeU0Cn1DjAYDhvhwKBHMOscmivPIClKgAOsREYzRYY=";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/s3rs/default.nix:11]:
```suggestion
    sha256 = "0sxjhqngpryzc6nxcrfmvfff4j8bfjj4zw57lqixal3v4kkh90lm";
```

**Comment by yanganto** [comment]
@SuperSandro2000  Thanks for the review.

**Pull request merged**